core/shark_turbine/ops/iree.py

```diff
@@ -50,29 +50,13 @@ def _emit_tensor_trace(kb: KernelBuilder, key: str, ts: list[Value]):
 
 @CustomOp.register(library=IREE_LIBRARY)
 class trace_tensor(CustomOp):
-    signature = "trace_tensor(str trace_key, Tensor tensor) -> ()"
+    signature = "trace_tensor(str trace_key, Tensor(a!) tensor) -> ()"
 
     def select(self, ksel: KernelSelection):
         ksel.attr_str(0)
-        ksel.arg_tensor(1)
+        ksel.arg_tensor(1, inplace_tied=True)
 
     def generate(self, ksel: KernelSelection, kb: KernelBuilder):
         key = cast(AttrArg, ksel.arg_descs[0])
         _emit_tensor_trace(kb, cast(str, key.v), [kb.arg_bindings[1]])
-        kb.yield_results()
-
-
-@CustomOp.register(library=IREE_LIBRARY)
```

Comment:
This trace_tensors op is still in the iree ops test, do we want to converge on trace_tensor and have it handle list targets?

Reply:
I'll need to remove it from the tests. Thanks. Unfortunately, new-torch has a TODO on this kind of tensor list on side-effecting ops, so I decided to remove it as the most practical way forward :/

We can have a helper that calls it multiple times (or whatever we want). I was also going to add a `functional_trace_tensor` which returns its value and works a bit better with the infra.

Reply:
Noted in the PR description.